deploy/deploy.ts

```diff
@@ -92,11 +109,120 @@ const deployFunction: DeployFunction = async function (hre: HardhatRuntimeEnviro
+  }
+   
+  // read in aave deployment lendingMarkets json file (https://docs.aave.com/developers/deployed-contracts/deployed-contracts)
+  let aaveAddressesArray
+  if(chainId == 1){
+    aaveAddressesArray = (JSON.parse(readFileSync("./aave/aaveMainnet.json", {encoding: "utf-8"}))).proto
+  }
+  if(chainId == 42){
+    aaveAddressesArray = (JSON.parse(readFileSync("./aave/aaveKovan.json", {encoding: "utf-8"}))).proto
+  }
+  else{
+    dim(`TestEnvironment! No deployed ATokens. Using Kovan as mock.`)
+    aaveAddressesArray = (JSON.parse(readFileSync("./aave/aaveKovan.json", {encoding: "utf-8"}))).proto
+  }
+
+  // we can filter here for aTokens that we want - by symbol
+  // const aTokenFilter = [aDai]
+
+  if(aTokenYieldSourceContract){ // needed check for typescript
+    green(`Now deploying aToken proxies`)
+
+    const aTokenYieldSourceInterface = new ethers.utils.Interface((await hre.artifacts.readArtifact("ATokenYieldSource")).abi)
+    let addressesProviderReg: string
+    if(chainId == 1){
+      addressesProviderReg = LENDING_POOL_ADDRESSES_PROVIDER_REGISTRY_ADDRESS_MAINNET
+    }
+    else {
+      addressesProviderReg = LENDING_POOL_ADDRESSES_PROVIDER_REGISTRY_ADDRESS_KOVAN
+    }
+
+    // deploy a proxy for each entry in aaveAddressesArray
+    for(const aTokenEntry of aaveAddressesArray){
+      
+      // if already deployed - skip
+      if(existsSync(`./deployments/${getChainByChainId(chainId).network}/${aTokenEntry.aTokenSymbol}.json`)){
+        dim(`${aTokenEntry.aTokenSymbol} already exists for this network`)
+        continue
+      }
+
+      let constructorArgs: string = aTokenYieldSourceInterface.encodeFunctionData(aTokenYieldSourceInterface.getFunction("initialize"),
+        [
+          aTokenEntry.aTokenAddress,
+          addressesProviderReg,
+          aTokenEntry.decimals,
+          `pt-${aTokenEntry.aTokenSymbol}`,
+          `pt${aTokenEntry.aTokenSymbol}`
+        ]
+      )
+
+      dim(`Creating Proxy for ${aTokenEntry.aTokenSymbol}`)
+      
+      const createATokenYieldSourceResult = await proxyFactoryContract.create(aTokenYieldSourceContract.address, constructorArgs) 
+     
+      // now generate deployments JSON entry -- need: address, abi (of instance), txHash, receipt, constructor args, bytecode
+      const receipt = await ethers.provider.getTransactionReceipt(createATokenYieldSourceResult.hash);
+      const createdEvent = proxyFactoryContract.interface.parseLog(receipt.logs[0]);
+
+      green(`aToken proxy for ${aTokenEntry.aTokenSymbol} created at ${createdEvent.args.created}`)
+
+      dim(`saving deployments file for ${aTokenEntry.aTokenSymbol}`)
+
+      let jsonObj: ProxyDeployment = {
+        address: createdEvent.args.created,
+        transactionHash: receipt.transactionHash,
+        receipt: receipt,
+        args: constructorArgs,
+        bytecode: `${prefix}${createdEvent.args.created}${postfix}`
```

Comment:
Looks like this is just concatenating the '0x' from the address in the middle.

I'd rather see you pull the bytecode directly; i.e. `provider.getCode(address)`